nimterop/types.nim

```diff
@@ -31,3 +31,31 @@ when defined(c):
 elif defined(cpp):
   type
     wchar_t* {.importc.} = object
+
+template enumOp*(op, typ, typout) =
+  proc op*(x: typ, y: int): typout {.borrow.}
+  proc op*(x: int, y: typ): typout {.borrow.}
+  proc op*(x, y: typ): typout {.borrow.}
+
+template defineEnum*(typ: untyped) =
```

Comment:
can we use `s/untyped/typed/` ? (always prefer the least powerful idiom)

Reply:
`typ: enum` doesn't work - Error: undeclared identifier: 'ENUM', neither does `typ: typed`. Am just removing it altogether for now.